tests/object_sync_test.go

```diff
@@ -1342,29 +1375,29 @@ func makeCustomVarTestData(t *testing.T) []*CustomVarTestData {
 			id + "-nested-dict":  `{"array":["answer?",42],"dict":{"another-key":"another-value","yet-another-key":4711},"top-level-entry":"good morning"}`,
 			id + "-nested-array": `[[1,2,3],{"contains-a-map":"yes","really?":true},-42]`,
 		},
-		VarsFlat: map[string]string{
-			id + "-array[0]":                         `foo`,
-			id + "-array[1]":                         `23`,
-			id + "-array[2]":                         `bar`,
-			id + "-dict.some-key":                    `some-value`,
-			id + "-dict.other-key":                   `other-value`,
-			id + "-string":                           `hello icinga`,
-			id + "-int":                              `-1`,
-			id + "-float":                            `13.37`,
-			id + "-true":                             `true`,
-			id + "-false":                            `false`,
-			id + "-null":                             `null`,
-			id + "-nested-dict.dict.another-key":     `another-value`,
-			id + "-nested-dict.dict.yet-another-key": `4711`,
-			id + "-nested-dict.array[0]":             `answer?`,
-			id + "-nested-dict.array[1]":             `42`,
-			id + "-nested-dict.top-level-entry":      `good morning`,
-			id + "-nested-array[0][0]":               `1`,
-			id + "-nested-array[0][1]":               `2`,
-			id + "-nested-array[0][2]":               `3`,
-			id + "-nested-array[1].contains-a-map":   `yes`,
-			id + "-nested-array[1].really?":          `true`,
-			id + "-nested-array[2]":                  `-42`,
+		VarsFlat: map[string]sql.NullString{
+			id + "-array[0]":                         toDBString(`foo`),
+			id + "-array[1]":                         toDBString(`23`),
+			id + "-array[2]":                         toDBString(`bar`),
+			id + "-dict.some-key":                    toDBString(`some-value`),
+			id + "-dict.other-key":                   toDBString(`other-value`),
+			id + "-string":                           toDBString(`hello icinga`),
+			id + "-int":                              toDBString(`-1`),
+			id + "-float":                            toDBString(`13.37`),
+			id + "-true":                             toDBString(`true`),
+			id + "-false":                            toDBString(`false`),
+			id + "-null":                             toDBString("null"),
+			id + "-nested-dict.dict.another-key":     toDBString(`another-value`),
```

Comment:
Not sure why you're changing the quotes around `null` here (and two more times below). Both do the same in this case.